pkg/reconciler/instances/istio/action.go

```diff
@@ -71,6 +71,27 @@ func (a *MainReconcileAction) Run(context *service.ActionContext) error {
 		return err
 	}
 
+	err = deployIstio(context, performer)
+
+	errPatchMutatingWebhook := performer.PatchMutatingWebhook(context.Context, context.KubeClient, context.Logger)
+
+	switch {
+	case err != nil && errPatchMutatingWebhook != nil:
+		errPatchMutatingWebhook = errors.Wrap(errPatchMutatingWebhook, "Could not patch MutatingWebhookConfiguration")
```

Comment:
Because you removed if-else statements now you have to wrap this error the same way in two cases which effects in code duplication. You could extract it to separate private function to build wrap error webhook with custom message.

Reply:
Based on your comment I agree additional if statement is required. Also wrap cannot be moved outside this place